accepted/0000-caasp-integration.md

```diff
@@ -0,0 +1,74 @@
+- Feature Name: CaaS Platform support
+- Start Date: 2019-09-24
+- RFC PR: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Initial integration for managing a CaaS Platform cluster from Uyuni/SUSE Manager.
+
+# Motivation
+[motivation]: #motivation
+
+Uyuni/SUSE Manager will offer the possibility to manage the nodes that comprise a CaaS Platform cluster.
+Uyuni/SUSE Manager will register and show the status of each node of the cluster.
+
+# Detailed design
+[design]: #detailed-design
+
+## Node status reporting
+
+All the nodes in the cluster can be registered to Uyuni/SUSE Manager using bootstrapping by the already known methods.
+The added benefit will be that the user can see from Uyuni/SUSE Manager if a node has packages updates and patch available, along with all the information that a Salt minion is presenting when registered to Uyuni/SUSE Manager (e.g. CVE auditing).
+Another benefit is that the user can associate staged channels to the nodes registered (along with an activation key) to associate channels to the nodes of the cluster.
+The agent `skuba-update`, running on each node of the cluster, automatically picks up the packages from the associated channels and proceed to automatically update the nodes.
+NOTE: `skuba-update` is **NOT** invoked by Uyuni/SUSE Manager and it is running autonomously on the nodes of the CaaS Platform cluster.
+
+For example: an organization could have a "staging" cluster with a set of staging channels. The operator would cherry-pick Kubernetes updates from the SCC channels into the staging ones and then wait for skuba-update to roll them out into the staging cluster. At a later time, once testing is green inside of the staging cluster, the packages can be promoted to the "production channels" and be rolled out to the production cluster.
+
+If minion blackout is feasible in an easy way, then we can think to introduce it.
+
+It is paramount to state in the documentation that this information is presented in a read-only fashion.
+If a user reboots the node, installs a patch or installs a package related to Kubernetes (`kubernetes-kubeadm kubernetes-kubelet kubernetes-client cri-o cni-plugins`) the Kubernetes may break.
+
+## Changes in the Kubernetes integration
+
+Under Virtual Host Manager, there is an already Kubernetes integration that supports CaaS Platform v3.
```

Comment:
Maybe I'd move this to earlier, below "Motivation" and start with: "_SUSE Manager 4.0 offers Kubernetes cluster integration. It works with CaaSP 3 (based on docker), with the following limitations:_
_- Cluster is presented as a system group_
_- Nodes are presented under Virtual Host Manager
_- Individual nodes can be managed and thus the cluster broken__
_- ..._

_This RFC is about extending that to support CaaSP4 and still supporting CaaSP3._"

If case this may work with some other Kubernetes distribution that we know about, I'd mention it, especially if it's a FLOSS one (for Uyuni). If not, I'd also mention it: "this RFC does not try to integrate other third-party Kubernetes distributions". Let's be very clear with the scope.

Reply:
OK